src/main/java/pl/javastart/task/TournamentStats.java

```diff
@@ -1,12 +1,64 @@
 package pl.javastart.task;
 
-import java.util.Scanner;
+import java.io.IOException;
+import java.util.*;
 
 public class TournamentStats {
 
+    private static final int SORT_BY_FIRST_NAME = 1;
+    private static final int SORT_BY_LAST_NAME = 2;
+    private static final int SORT_BY_RESULT = 3;
+    private static final int SORTING_BY_REVERSE = 2;
+    private static final int SORT_ASCENDING = 1;
+    private static boolean reverseList;
+
     void run(Scanner scanner) {
-        // tutaj dodaj swoje rozwiązanie
-        // użyj przekazanego scannera do wczytywania wartości
+        List<TournamentPlayer> tournamentPlayersList = createListOfPersons(scanner);
+        Comparator<TournamentPlayer> comparator = tournamentPlayerComparator(scanner);
+        tournamentPlayersList.sort(comparator);
+        try {
+            TournamentStatsWriter tournamentStatsWriter = new TournamentStatsWriter();
+            tournamentStatsWriter.saveTournamentStats(tournamentPlayersList);
+        } catch (IOException e) {
+            System.err.println("Statystyki nie zostały zapisane");
+        }
+    }
+
+    public List<TournamentPlayer> createListOfPersons(Scanner scanner) {
+        List<TournamentPlayer> listOfPlayers = new LinkedList<>();
+        String line;
+        do {
+            System.out.println("Podaj wynik kolejnego gracza (lub stop):");
+            line = scanner.nextLine();
+            if (line.equalsIgnoreCase("stop")) {
+                break;
+            }
+            TournamentPlayer player = new TournamentPlayer(line);
+            listOfPlayers.add(player);
+        } while (!line.equals("stop"));
```

Comment:
Jeżeli w ifie masz breaka, to ten warunek w pętli nie do końca ma sens, szczególnie, że raz używasz equals, a drugim razem equalsIgnoreCase. Jeżeli już, to można to zapisać z flagą boolean
```
boolean exit = false;
do {
...
  if (line.equalsIgnoreCase("stop")) {
    exit = true;
  }
while (!exit);
```
Zmienną line można też wtedy zadeklarować wewnątrz pętli